src/pages/en/guides/content-collections.mdx

```diff
@@ -121,6 +121,22 @@ If you **do not** already extend Astro's `strict` or `strictest` recommended Typ
 }
 ```
 
+:::note
+When using JavaScript files in an Astro project using a `tsconfig.json`, you won't get intellisense or type-checking, unless your project is using a `jsconfig.json`. As such, when using a JavaScript file as your Content Collections configuration file, you won't get those benifits.
+To get type checking and intellisense for JavaScript files in that case, you will need to update your `tsconfig.json` to enable the `allowJs` compiler option.
+
+```json title="tsconfig.json" ins={6}
+{
+  // Note: No change needed if you use "astro/tsconfigs/strict" or "astro/tsconfigs/strictest"
+  "extends": "astro/tsconfigs/base",
+  "compilerOptions": {
+    "strictNullChecks": true,
+    "allowJs": true
+  }
+}
+```
+:::
```

Comment:
Thanks again for this @MoustaphaDev! I read through the page thinking about this in context and I actually think this is maybe the right place for this. What do you think about these tweaks?

My goals here were:
- slim this down a bit to be clear and compact
- frame positively: doing X? => do Y

```suggestion
:::note[Not using TypeScript?]
If you use `.js` or `.mjs` files in an Astro project, you can enable IntelliSense and type checking in your editor using a [`jsconfig.json`](https://code.visualstudio.com/docs/languages/jsconfig) or by enabling `allowJs` in your `tsconfig.json`:

```json title="tsconfig.json" ins={5}
{
  "extends": "astro/tsconfigs/base",
  "compilerOptions": {
    "strictNullChecks": true,
    "allowJs": true
  }
}
```
:::
```

Reply:
I would recommend `allowJs` first over `jsconfig.json`, since the latter isn't standard and not every tool load it / load it properly

Reply:
Thanks @Princesseuh! In that case I might be tempted to cut it down to just this. No extra action needed if you have `jsconfig.json` but we’re not recommending that path anyway.

````suggestion
:::note[Not using TypeScript?]
If you use `.js` or `.mjs` files in an Astro project, you can enable IntelliSense and type checking in your editor by enabling `allowJs` in your `tsconfig.json`:
```json title="tsconfig.json" ins={5}
{
  "extends": "astro/tsconfigs/base",
  "compilerOptions": {
    "strictNullChecks": true,
    "allowJs": true
  }
}
```
:::
````

Reply:
Sounds good